drivers/usb/device/usb_dc_stm32.c

```diff
@@ -396,12 +396,15 @@ static int usb_dc_stm32_init(void)
 #endif /* CONFIG_USB_DEVICE_SOF */
 
 #if defined(CONFIG_SOC_SERIES_STM32H7X)
-	/* Currently assuming FS mode. Need to disable the ULPI clock on USB2.
-	 * Need to make this dependent on HS or FS config.
+#if defined LL_AHB1_GRP1_PERIPH_USB2OTGHSULPI
+	/* The USB2 controller only works in FS mode, but we still need to
+	 * disable its ULPI clock. As the USB1 controller is an HS one, this is
+	 * done in the common path.
```

Comment:
Can we somehow use device tree to get this information ?

Reply:
> Can we somehow use device tree to get this information ?

It's not really clear to me. The USB controller is declared as a `st,stm32-otgfs` in the device tree because it can't support HS physically, but from the register point of view it is actually a HS controller with an ULPI interface.

Reply:
I realized that this patch mixes unrelated comment fixes and code changes. I have therefore split this one in two parts:
- One that fixes the existing comment about the second USB controller, but doesn't change any code. The part _Need to make this dependent on HS or FS config_ is wrong as the second controller can only work in FS mode with the internal FS PHY.
- One that limit the changes on the second USB controller to the SoCs that have two USB controllers.

Reply:
The problem there is to make the link between the USB controller and the USB PHY. That something that can be modeled in the device tree, however what we want here is to **disable** the **not used** PHY clock, while the device tree models the devices that are used.

After thinking a bit more I have found a way to use the device tree, by checking for the `st-stm32-otgfs` compat property of the USB controller. That said it hardcodes in a bit more places that the USB1 controller is HS and the USB2 controller is FS on STM32H7x SoC. That will break if ST decides to create a STM32H7x SoC with 2 HS controllers.